src/components/OrderPanel/BookingDatesForm/BookingDatesForm.js

```diff
@@ -373,6 +384,83 @@ const showPreviousMonthStepper = (currentMonth, timeZone) => {
   return isDateSameOrAfter(prevMonthDate, currentMonthDate);
 };
 
+// return a list of timeslots that exist between startDate and endDate
+const filterTimeSlotsByDate = (allTimeSlots, startDate, endDate) => {
+  return Object.values(allTimeSlots).filter(
+    ({ attributes: { start, end } }) =>
+      // Check if the timeslot is within or overlaps with the selected dates
+      (start < startDate && end > startDate) ||
+      (start >= startDate && end <= endDate) ||
+      (start < endDate && end > endDate)
+  );
+};
+
+// Finds the timeslot with the smallest number of seats
+const findMinSeatsTimeSlot = timeSlots => {
+  return timeSlots.reduce((minSeatsSlot, timeSlot) => {
+    const { seats } = timeSlot.attributes;
+    return !minSeatsSlot || seats < minSeatsSlot.seats ? timeSlot.attributes : minSeatsSlot;
+  }, null);
+};
+
+// Main function to get the seat options based on the minimum seats available in the date range
+const getMinSeatsOptions = (allTimeSlots, startDate, endDate) => {
+  const filteredTimeSlots = filterTimeSlotsByDate(allTimeSlots, startDate, endDate);
+  const minSeatsSlot = findMinSeatsTimeSlot(filteredTimeSlots);
+
+  // Return the array of seat options from 1 to the minimum seats available, capped at 100
+  const maxOptions = 100;
+  return minSeatsSlot
+    ? Array.from({ length: Math.min(minSeatsSlot.seats, maxOptions) }, (_, i) => i + 1)
+    : [];
+};
+
+// Checks if two timeslots are consequtive
+const areConsecutiveTimeSlots = (timeSlotA, timeSlotB) =>
+  new Date(timeSlotA.attributes.end).getTime() === new Date(timeSlotB.attributes.start).getTime();
+
+// Find the index of a the first consecutive timeslot in a list of timeslots
+const findIndexOfFirstConsecutiveTimeSlot = (timeSlots, index) =>
+  index > 0 && areConsecutiveTimeSlots(timeSlots[index - 1], timeSlots[index])
+    ? findIndexOfFirstConsecutiveTimeSlot(timeSlots, index - 1)
+    : index;
+
+// find the index of the last consecutive timeslot in a list of timeslots
+const findIndexOfLastConsecutiveTimeSlot = (timeSlots, index) =>
+  index < timeSlots.length - 1 && areConsecutiveTimeSlots(timeSlots[index], timeSlots[index + 1])
+    ? findIndexOfLastConsecutiveTimeSlot(timeSlots, index + 1)
+    : index;
+
+// Find and combine adjacent/consecutive timeslots into one timeslot
+const combineConsecutiveTimeSlots = (slots, startDate) => {
+  // Locate the index of the timeslot containing startDate
+  const startIndex = slots.findIndex(({ attributes }) => {
+    const { start, end } = attributes;
+    const startTime = new Date(start).getTime();
+    const endTime = new Date(end).getTime();
+    return startDate.getTime() >= startTime && startDate.getTime() < endTime;
+  });
+
+  // Return null if no timeslot matches startDate
+  if (startIndex === -1) return null;
```

Comment:
Why null and not an empty array? If this is used in the place where array is expected, it can cause an error (`arr.map(/*something*/)` => `null.map(/*something*/)`)